### PR TITLE
Refactor machine health checks to use cluster and kubernetes.Object

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*

*Description of changes:*
Refactor machine health checks to use cluster and kubernetes.Object. Also convert templater method that converts objects to yaml a generic method.

*Testing (if applicable):*
- unit tests
- manual tests

*Documentation added/planned (if applicable):*
N/A

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/cluster/config_manager_test.go [hunk 1]
```diff
@@ -219,9 +219,14 @@ func TestConfigManagerMissingAPIVersion(t *testing.T) {
 		},
 	}
 
-	yamlManifest, err := templater.ObjectsToYaml(wantCluster, wantDockerDatacenter)
+	yamlManifest, err := templater.ObjectsToYaml(wantCluster)
 	g.Expect(err).To(BeNil())
 
+	datacenterYamlManifest, err := templater.ObjectsToYaml(wantDockerDatacenter)
```
Comment (g-gaston) [inline pkg/cluster/config_manager_test.go:225]: curious why is this changing?

Comment (mitalipaygude) [reply pkg/cluster/config_manager_test.go]: changed it back.

File: pkg/clients/kubernetes/runtimeclient.go [hunk 2]
```diff
@@ -76,3 +76,13 @@ type restConfigurator func([]byte) (*rest.Config, error)
 func (c restConfigurator) Config(data []byte) (*rest.Config, error) {
 	return c(data)
 }
+
+// ObjectsToRuntimeObjects converts generic objects to runtime objects.
+func ObjectsToRuntimeObjects[T client.Object](objs []T) []runtime.Object {
```
Comment (g-gaston) [inline pkg/clients/kubernetes/runtimeclient.go:81]:
```suggestion
func ObjectsToRuntimeObjects[T runtime.Object](objs []T) []runtime.Object {
```

File: pkg/clients/kubernetes/runtimeclient.go [hunk 3]
```diff
@@ -76,3 +76,13 @@ type restConfigurator func([]byte) (*rest.Config, error)
 func (c restConfigurator) Config(data []byte) (*rest.Config, error) {
 	return c(data)
 }
+
+// ObjectsToRuntimeObjects converts generic objects to runtime objects.
```
Comment (g-gaston) [inline pkg/clients/kubernetes/runtimeclient.go:80]:
```suggestion
// ObjectsToRuntimeObjects converts objects of another type to runtime.Object's.
```

**Review by g-gaston — APPROVED** [review]

**Pull request merged**